apps/emqx/src/emqx_socket_connection.erl

```diff
@@ -362,12 +362,24 @@ run_loop(
         {ok, Data} ->
             NState = start_idle_timer(State),
             handle_recv({recv_more, Data}, Parent, NState);
-        {select, _SelectInfo} ->
+        {completion, {completion_info, _, _}} ->
```

Comment:
Q: Are those changes because of Dialyzer complaints?

Since:
1. Windows [is not supported](https://github.com/emqx/emqx/blob/483d4bc1ec16fbbaca4f6e803058d7aa6b7b677e/apps/emqx/src/emqx_schema.erl#L639-L640) currently.
2. Impossible to get `select_read` since it needs to be explicitly enabled on the socket.
3. Should be impossible to get partial data because we explicitly request 0 bytes.

Otherwise, extra match on `recv` is pointless since we do not call any other `socket` functions apart from `socket:recv/4` here.